File: Upcoder/Lập trình cơ bản/7.Other/TINH_TONG/LTCB-8.cpp
#include <bits/stdc++.h>
using namespace std;

long double Dequy(long long n)
{
    long double result = 1.0 / 2.0 + 3.0 / 4.0;
    for (int i = 2; i <= n; ++i) {
        result += (2 * i + 1.0) / (2 * i + 2.0);  
    }
    return result;
}

int main()
{
    long long n;
    cin >> n;
    if(n == 0)
        cout << 1.0 / 2.0;
    else
        cout << Dequy(n); 
    return 0;
}

File: Upcoder/Lập trình cơ bản/7.Other/TINH_TONG/LTCB-8.py
def Dequy(n):
    result = 1.0 / 2.0 + 3.0 / 4.0
    for i in range(2, n + 1):
        result += (2 * i + 1.0) / (2 * i + 2.0)
    return result

n = int(input())
if n == 0:
    print(1.0 / 2.0)
else:
    print(Dequy(n), end="")
